File: Simple Linear Regression.py
import pandas as pd
import matplotlib.pyplot as plt

# Importing datase
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:, ["YearsExperience"]].values
y = dataset.iloc[:, 1].values

# Visualizing data
plt.scatter(X, y)
plt.show()

# Values can be represented by a Simple Linear Regression Model

# Splitting dataset to Training Set and Test Set
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

# Fitting values to the model
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predicting values from the test set
y_pred = regressor.predict(X_test)

# Visualising Training Set with the model
plt.scatter(X_train, y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience - Training Set')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

# Visualsing Test Set with the model
plt.scatter(X_test, y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience - Test Set')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

# Model give a good fit

# Training model with the whole dataset
regressor.fit(X, y)

File: Simple Linear Regression.R
#install.packages("ggplot2")
library(ggplot2)

# Importing dataset
dataset <- read.csv("Salary_Data.csv")

#Visualising Dataset to see linear trend
ggplot(dataset, aes(YearsExperience, Salary)) + geom_point()

# Splitting dataset into Training and Test sets
#install.packages("caTools")
library(caTools)
set.seed(123)
split <- sample.split(dataset$Salary, SplitRatio = 0.8)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fitting training set to Linear Regression Model
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)

# Predicting on test set
y_pred = predict(regressor, newdata = test_set)

# Visualising linear model with Training Set
ggplot() + geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), colour = 'blue') +
  ggtitle('Salary vs Experience (Training set)') + xlab('Years of experience') + ylab('Salary')
  
# Visualising linear model with Test Set
ggplot() + geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)), colour = 'blue') +
  ggtitle('Salary vs Experience (Test set)') + xlab('Years of experience') + ylab('Salary')

# Training the model on complete dataset
regressor = lm(formula = Salary ~ YearsExperience, data = dataset)
